pycqed/measurement/multi_qubit_module.py

```diff
@@ -1843,6 +1843,101 @@ def measure_measurement_induced_dephasing(qb_dephased, qb_targeted, phases, amps
+        Sweep pulse cancellation parameters and measure Ramsey on cancelled qubits.
+        The sweep point keys should be of the form `qb.param`, where `qb` is the
+        name of the qubit the cancellation if for and `param` is a parameter in
+        the pulses cancellation_params dict.
+
+        For example to sweep the amplitude of the cancellation pulse on qb1,
+        you could configure the sweep points as `SweepPoints('qb1.amplitude',
+        np.linspace(0, 1, 21))`.
+
+        The second sweep dimension of sweep_points must be called 'phases'.
+        """
+
+        if isinstance(driven_qubit, str):
+            driven_qubit = dev.get_qb(driven_qubit)
+        ramsey_qubits = [dev.get_qb(qb) if isinstance(qb, str) else qb
+                         for qb in ramsey_qubits]
+        ramsey_qubit_names = [qb.name for qb in ramsey_qubits]
+
+        MC = dev.instr_mc.get_instr()
+        if label is None:
+            label = f'drive_{driven_qubit.name}_cancel_'\
+                    f'{list(sweep_points[0].keys())}'
+
+        if prep_params is None:
+            prep_params = dev.get_prep_params(ramsey_qubits)
+
+        sweep_points.add_sweep_dimension()
+        sweep_points.add_sweep_parameter('phase', phases, 'deg', 'Ramsey phase')
+        if exp_metadata is None:
+            exp_metadata = {}
+
+        for qb in [driven_qubit] + ramsey_qubits:
+            qb.prepare(drive='timedomain')
+
+        cal_states = CalibrationPoints.guess_cal_states(cal_states,
+                                                        for_ef=False)
+        cp = CalibrationPoints.multi_qubit(
+            [qb.name for qb in ramsey_qubits], cal_states,
+            n_per_state=n_cal_points_per_state)
+        operation_dict = dev.get_operation_dict()
+
+        seq, sweep_vals = mqs.drive_cancellation_seq(
+            driven_qubit.name, ramsey_qubit_names, operation_dict, sweep_points,
+            pulse=pulse, n=n, prep_params=prep_params, cal_points=cp,
+            upload=False)
+
+        [seq.repeat_ro(f"RO {qbn}", operation_dict) for qbn in ramsey_qubit_names]
+
+        sweep_func = awg_swf.SegmentHardSweep(
+                sequence=seq, upload=upload,
+                parameter_name='segment_index')
+        MC.set_sweep_function(sweep_func)
+        MC.set_sweep_points(sweep_vals)
+
+        det_func = get_multiplexed_readout_detector_functions(
+            ramsey_qubits,
+            nr_averages=max([qb.acq_averages() for qb in ramsey_qubits]))\
+            ['int_avg_det']
+        MC.set_detector_function(det_func)
+
+        sweep_points_for_analysis = [
+            {k: (np.repeat(v[0], len(phases)), v[1], v[2])
+             for k, v in sweep_points[0].items()}
+        ]
+        len_sweep = len(list(sweep_points[0].values())[0][0])
+        sweep_points_for_analysis[0]['phase'] = \
+            (np.tile(phases, len_sweep), 'rad', 'Ramsey phase')
+        meas_obj_sweep_points_map = {qbn: ['phase'] for qbn in ramsey_qubit_names}
```

Comment:
@stephlazar might have a comment on this when reviewing the analysis part.

Reply:
Yes, this won't work with the fix that I added last week. The concept for SweepPoints is that they should always describe the experiment conceptually. So if your experiment is a 2D sweep conceptually, then the SweepPoints should have 2 dimensions. And the meas_obj_sweep_points_map for each measured object should have at least one key from each dimension. I will try to fix things myself and will get in touch with you @antsr if I can't figure it out.

Reply:
Fixed now.